solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -2580,57 +2596,105 @@ sp_api::impl_runtime_apis! {
 pub struct OwnApplySession;
 impl tanssi_initializer::ApplyNewSession<Runtime> for OwnApplySession {
     fn apply_new_session(
-        changed: bool,
+        _changed: bool,
         session_index: u32,
-        all_validators: Vec<(AccountId, ValidatorId)>,
-        queued: Vec<(AccountId, ValidatorId)>,
+        all_validators: Vec<(AccountId, nimbus_primitives::NimbusId)>,
+        _queued: Vec<(AccountId, nimbus_primitives::NimbusId)>,
     ) {
-        use frame_support::traits::OneSessionHandler;
-        let all_validators_initializer: Vec<(&AccountId, ValidatorId)> = all_validators
-            .iter()
-            .map(|(validator, key)| (validator, key.clone()))
-            .collect();
-        let queued_initializer: Vec<(&AccountId, ValidatorId)> = queued
-            .iter()
-            .map(|(validator, key)| (validator, key.clone()))
-            .collect();
-
-        Initializer::on_new_session(
-            changed,
-            all_validators_initializer.into_iter(),
-            queued_initializer.into_iter(),
-        );
+        // Order is same as in tanssi
+        // 1.
         // We first initialize Configuration
         CollatorConfiguration::initializer_on_new_session(&session_index);
-        // Next: Registrar
+        // 2. Second, registrar
         ContainerRegistrar::initializer_on_new_session(&session_index);
-        // // Next: AuthorityMapping
-        // AuthorityMapping::initializer_on_new_session(&session_index, &all_validators);
 
-        // let next_collators = queued.iter().map(|(k, _)| k.clone()).collect();
+        // 3. AuthorityMapping
+        TanssiAuthorityMapping::initializer_on_new_session(&session_index, &all_validators);
```

Comment:
This is wrong, the authority mapping should only be done for collators, i.e. invulnerables. So let's fetch the invulnerable info first

Reply:
Fixed in 0e9953bc54d2eb955fda6b0031cffa6582fddc7e